apis/kueue/v1beta1/provisioningrequestconfig_types.go

```diff
@@ -57,6 +58,58 @@ type ProvisioningRequestConfigSpec struct {
 	// +listType=set
 	// +kubebuilder:validation:MaxItems=100
 	ManagedResources []corev1.ResourceName `json:"managedResources,omitempty"`
+
+	// retryStrategy defines strategy for retrying ProvisioningRequest
+	// if nil then default configuration will be applied
+	// to switch off retry mechanism completely, set retryStrategy.backoffLimitCount to 0.
+	//
+	// +optional
+	RetryStrategy *ProvisioningRequestRetryStrategy `json:"retryStrategy,omitempty"`
+}
+
+const (
+	DefaultBackoffLimitCount  int32 = 3
+	DefaultBackoffBaseSeconds int32 = 60   // 1 min
+	DefaultBackoffMaxSeconds  int32 = 1800 // 30 min
+)
+
+var (
+	DefaultRetryStrategy = ProvisioningRequestRetryStrategy{
```

Comment:
Maybe this could have its own type, so that we don't use ptrs inside? Just
```
type defualtRetryStrategy struct {
  int32 backoffLimitCount
  ...
```
then we could refer to the defaults via the `DefaultRetryStrategy`. Cause currently we have two sources of truth when it comes to the defaults (the structure and the constants).

Reply:
We discussed it offline and come to conclusion that it would be the best to get rid of some pointers in the API and default them with kubebuilder annotation. `BackOffLimitCount` should be a pointer thought, so when it's null, retries will happen endlessly